File: src/dataframe/rust/stats/statistical_tests/shapiro_wilk.test.R
#!/usr/bin/env Rscript

# R script for Shapiro-Wilk test comparisons
# Usage: Rscript shapiro_wilk.test.R <function_name> <args...>

args <- commandArgs(trailingOnly = TRUE)

if (length(args) < 1) {
  stop("Function name required")
}

function_name <- args[1]

if (function_name == "shapiro.test.normality") {
  # Shapiro-Wilk test
  data_str <- gsub("^\\[|\\]$", "", args[2])
  data <- as.numeric(strsplit(data_str, ",")[[1]])
  
  # Perform Shapiro-Wilk test
  result <- shapiro.test(data)
  
  cat(jsonlite::toJSON(list(
    test_statistic = result$statistic,
    p_value = result$p.value,
    reject_null = result$p.value < 0.05
  ), auto_unbox = TRUE))
  
} else {
  stop(paste("Unknown function:", function_name))
}

File: packages/testing/statistical_tests/spot_checks/shapiro-wilk.test.ts
import { stats } from "@tidy-ts/dataframe";

Deno.test("Shapiro-Wilk Normality Test Spot Check", () => {
  console.log("\n📊 Shapiro-Wilk Normality Test");
  console.log("-".repeat(80));

  const data = [2.5, 3.1, 2.8, 3.3, 2.9, 3.0, 2.7, 3.2];

  const result = stats.test.normality.shapiroWilk({ data });

  console.log("Data:", data);
  console.log("W-statistic:", result.testStatistic.value);
  console.log("p-value:", result.pValue);
});

console.log("\n" + "=".repeat(80));
console.log("SHAPIRO-WILK NORMALITY TEST SPOT CHECK");
console.log("=".repeat(80));
